Packs/AWS-AccessAnalyzer/Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py

```diff
@@ -309,24 +316,81 @@ def test_function():
         return return_error(str(e))
 
 
+def fetch_incidents(last_run: dict = None):
+    dParams = demisto.params()
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('time') if isinstance(last_run, dict) else None
+
+    # Handle first time fetch, fetch incidents retroactively
+    if last_fetch is None:
+        today = datetime.today()
+        delta = today - timedelta(days=50)
+        last_fetch = date_to_timestamp(delta)
+
+    incidents: list = []
+    dArgs = {}
+    dArgs['roleArn'] = dParams.get('roleArn')
+    dArgs['region'] = dParams.get('region')
+    dArgs['roleSessionName'] = dParams.get('roleSessionName')
+    dArgs['roleSessionDuration'] = dParams.get('roleSessionDuration')
+    dArgs['analyzerArn'] = dParams.get('analyzerArn')
+    dArgs['maxResults'] = 25
+
+    nextToken = None
+    incidents = []
+    tmp_last_fetch = last_fetch
+
+    while True:
+
+        if nextToken:
+            dArgs['nextToken'] = nextToken
+        raw_incidents = list_findings_command(dArgs)
+
+        for raw_incident in raw_incidents[0]['findings']:
+            incident = {
+                'name': f"AWS Access Analyzer Alert - {raw_incident['id']}",
+                'occurred': raw_incident['updatedAt'] + 'Z',
+                'rawJSON': json.dumps(raw_incident)
+            }
+
+            inc_timestamp = date_to_timestamp(raw_incident['updatedAt'])
+
+            if inc_timestamp > int(last_fetch):
+                incidents.append(incident)
+                if inc_timestamp > tmp_last_fetch:
+                    tmp_last_fetch = inc_timestamp
+            else:
+                break
+
+        if 'nextToken' in raw_incidents[0]:
+            nextToken = raw_incidents[0]['nextToken']
+        else:
+            break
+
+    demisto.setLastRun({"time": tmp_last_fetch if tmp_last_fetch else last_fetch})
+    return incidents
+
+
 """EXECUTION BLOCK"""
-try:
-    if demisto.command() == 'test-module':
-        result = test_function()
-    elif demisto.command() == 'aws-access-analyzer-list-analyzers':
-        list_analyzers_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-list-analyzed-resource':
-        list_analyzed_resource_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-list-findings':
-        list_findings_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-get-analyzed-resource':
-        get_analyzed_resource_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-get-finding':
-        get_finding_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-start-resource-scan':
-        start_resource_scan_command(demisto.args())
-    elif demisto.command() == 'aws-access-analyzer-update-findings':
-        update_findings_command(demisto.args())
-
-except Exception as e:
-    return_error(f"Error has occured in AWS Access Analyzer Integration: {str(e)}")
```

Comment:
why was this removed?

Reply:
This try and catch is around the entire block of code that is executed for this integration, so it made it difficult to debug errors as they are always reported as coming from the exception block. This is opposed to reporting the exact line of code that threw the error.  Might be an improvement to consider wrapping each API call in a try catch instead of 1 global one.  Essentially I removed it to make debugging easier. I can add this back if you think it is best here.

Reply:
It is best as it was so the user does not get the traceback in the error when they run the command.

Reply:
👍